lib/api/core/validation/index.js

```diff
@@ -1,690 +1,708 @@
+                  return null;
+                });
+            });
 
-                return null;
-              });
+            promises.push(promise);
           });
-
-          promises.push(promise);
         });
+
+        return Bluebird
+          .all(promises)
+          .then(() => {
+            this.specification = specification;
+            return this.kuzzle.pluginsManager.trigger('log:info', 'All validation specifications have been treated.');
+          });
       });
+  }
 
-      return Promise
-        .all(promises)
-        .then(() => {
-          this.specification = specification;
-          return this.kuzzle.pluginsManager.trigger('log:info', 'All validation specifications have been treated.');
+  /**
+   * @param {string} indexName
+   * @param {string} collectionName
+   * @param {CollectionSpecification} collectionSpec
+   * @param {boolean} verboseErrors
+   * @returns {Promise<boolean>}
+   */
+  isValidSpecification(indexName, collectionName, collectionSpec, verboseErrors) {
+    // We make a deep clone to avoid side effects
+    const specification = _.cloneDeep(collectionSpec);
+    verboseErrors = verboseErrors || false;
+
+    return new Bluebird(resolve => {
+      this
+        .curateCollectionSpecification(indexName, collectionName, specification, true, verboseErrors)
+        .then(result => {
+          if (verboseErrors) {
+            if (result.isValid === false) {
+              return resolve(result);
+            }
+          }
+          return resolve({isValid: true});
+        })
+        .catch(error => {
+          // we do not want to reject since this method goal is to know what is going wrong with the given spec
+          return resolve({isValid: false, errors: [error]});
         });
     });
-};
+  }
 
-/**
- * @param {string} indexName
- * @param {string} collectionName
- * @param {CollectionSpecification} collectionSpec
- * @param {boolean} verboseErrors
- * @returns {Promise<boolean>}
- */
-Validation.prototype.isValidSpecification = function validationIsValidSpecification (indexName, collectionName, collectionSpec, verboseErrors) {
-  // We make a deep clone to avoid side effects
-  const specification = _.cloneDeep(collectionSpec);
-  verboseErrors = verboseErrors || false;
-
-  return new Promise (resolve => {
-    this
-      .curateCollectionSpecification(indexName, collectionName, specification, true, verboseErrors)
-      .then(result => {
+  /**
+   * @param {string} indexName
+   * @param {string} collectionName
+   * @param {CollectionSpecification} collectionSpec
+   * @param {boolean} dryRun
+   * @param {boolean} verboseErrors
+   * @returns {CollectionSpecification}
+   * @rejects InternalError
+   */
+  curateCollectionSpecification(indexName, collectionName, collectionSpec, dryRun, verboseErrors) {
+    var
+      errorMessage = '',
+      result,
+      errorObject;
+
+    dryRun = dryRun || false;
+    verboseErrors = verboseErrors || false;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "verboseErrors". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1226)